app/src/main/java/com/example/mytestproject/ui/weatherData/WeatherDataViewModel.kt

```diff
@@ -16,21 +17,35 @@ class WeatherDataViewModel : ViewModel() {
 
     private var disposable: Disposable? = null
 
-    private val _weatherDataApi: MutableLiveData<WeatherData> =
+    var viewState  = MutableLiveData<WeatherViewState> ()
```

Comment:
You need to add _viewState: LiveData as it was earlier. 
Only this LiveData should be visible outside of ViewModel. And MutableLiveData should be private.